src/components/pages/import.js

```diff
@@ -1,152 +1,149 @@
 import { CloudDownloadOutlined, CloudSyncOutlined } from '@ant-design/icons';
 
-export default function Import() {
-  const { t } = useTranslation('import');
+const { Panel } = Collapse;
 
-  const { language } = useLanguage();
+function Import({ initialState }) {
+  const { t } = useTranslation('import');
   const { formatDate } = useDateFormat();
-  const clientConfig = useService(ClientConfig);
-  const importApiClient = useService(ImportApiClient);
-  const languageNameProvider = useService(LanguageNameProvider);
-
-  const sourceMenuItems = clientConfig.importSources;
-  const [selectedSource, setSelectedSource] = useState(null);
-  const [selectedDocumentKeys, setSelectedDocumentKeys] = useState([]);
-  const [importableDocuments, setImportableDocuments] = useState([]);
-  const [isFetchingImportableDocuments, setIsFetchingImportableDocuments] = useState(false);
-  const [isCreatingNewImportBatch, setIsCreatingNewImportBatch] = useState(false);
-
-  const handleImportClick = async () => {
-    const selectedDocs = selectedDocumentKeys
-      .map(key => importableDocuments.find(doc => doc.key === key));
 
-    const batch = {
-      hostName: selectedSource.hostName,
-      documentsToImport: selectedDocs
-    };
+  const { batch } = initialState;
+  const { hostName, allowUnsecure } = batch.batchParams;
 
-    try {
-      setIsCreatingNewImportBatch(true);
-      const result = await importApiClient.postImportBatch(batch);
-      // eslint-disable-next-line no-console
-      console.log('Batch created', result.batch);
-    } catch (error) {
-      handleApiError({ error, t });
-    } finally {
-      setIsCreatingNewImportBatch(false);
-    }
+  const renderStatus = processed => {
+    return processed
+      ? <span>{t('taskStatusDone')}</span>
+      : <span>{t('taskStatusPending')}</span>;
   };
 
-  const handleSourceMenuChange = async value => {
-    setIsFetchingImportableDocuments(true);
-    const newSelectedSource = sourceMenuItems.find(source => source.name === value);
-    setSelectedSource(newSelectedSource);
-    setImportableDocuments([]);
-    setSelectedDocumentKeys([]);
-
-    try {
-      const { documents } = await importApiClient.getImports(newSelectedSource.hostName);
-      setImportableDocuments(documents);
-    } catch (error) {
-      handleApiError({ error, t });
+  const renderBatchStatus = ({ progress }) => {
+    if (progress === 1) {
+      return <span>{t('batchStatusDone')}</span>;
+    }
+    if (progress === 0) {
+      return <span>{t('batchStatusPending')}</span>;
     }
-    setIsFetchingImportableDocuments(false);
+    return <span>{t('batchStatusProcessing', { progress: batch.progress })}</span>;
   };
 
-  const renderImportType = doc => {
+  const renderDate = date => <span>{formatDate(date)}</span>;
+  const renderTitle = ({ title, slug }) => (
+    <a
+      target="_blank"
+      href={getImportedArticleUrl({ hostName, allowUnsecure, slug })}
+      rel="noreferrer"
```

Comment:
Tiny thing, we could throw in `noopener` too, I know the linter prompted for `noreferrer` but then I'd say have both to be consistent with other links.

Reply:
Added, thanks!